runtime/src/main/java/org/corfudb/runtime/view/stream/AddressMapStreamView.java

```diff
@@ -395,21 +393,17 @@ protected long getMaxGlobalFromMaxEntries(int maxEntries) {
         StreamAddressSpace streamAddressSpace = runtime.getSequencerView()
                 .getStreamAddressSpace(new StreamAddressRange(getId(), Address.MAX, getCurrentGlobalPosition()));
 
-        long size = streamAddressSpace.getAddressMap().getLongCardinality();
+        long size = streamAddressSpace.size();
         if (size == 0L) {
             return Address.NON_ADDRESS;
         }
 
+        // TODO(Maithem): probably can be removed, optimization not needed
         if (size <= maxEntries) {
             return streamAddressSpace.getHighestAddress();
         }
 
-        LongIterator it = streamAddressSpace.getAddressMap().getLongIterator();
-        while (--maxEntries > 0) {
-            it.next();
-        }
-
-        return it.next();
+        return streamAddressSpace.select(maxEntries);
```

Comment:
Is there an off by one error here?

Reply:
+1:

    @Test
    public void xyz() {
        StreamAddressSpace space = new StreamAddressSpace(Arrays.asList(1L, 2L, 6L, 8L, 12L, 15L));
        System.out.println(">>>> " + space.select(space.size()));
    }

Reply:
Good catch. Will add more tests.